### PR TITLE
Add xml.sax.saxutils.XMLGenerator parameter short_empty_elements.

### LINKED ISSUE #1264: xml.sax.saxutils.XMLGenerator missing
https://github.com/python/typeshed/blob/master/stdlib/2and3/xml/sax/saxutils.pyi#L15 presents ``__init__`` with parameters
```py
def __init__(self, out=..., encoding=...) -> None: ...
```
although in reality [the definition for Python 3.4](https://github.com/python/cpython/blob/master/Lib/xml/sax/saxutils.py#L115) is
```py
__init__(self, out=None, encoding='iso-8859-1', short_empty_elements=False)
```
So, ``short_empty_elements`` is missing.

### PR DESCRIPTION
**Pull request by mcepl**
Fix #1264

### CODE REVIEW

**Comment by mcepl** [comment]
Damn, this is wrong. It should be ``py3k`` only.

**Comment by JelleZijlstra** [comment]
Can you fix the lint error?

**Comment by JelleZijlstra** [comment]
Also, I don't think this is reason to move xml out of 2and3. We should instead do something like `if sys.version_info > (3, 0): def __init__(..., short_empty_elements): ... else: def __init__(...): ...`.

**Comment by mcepl** [comment]
> Also, I don't think this is reason to move xml out of 2and3. We should instead do something like if sys.version_info > (3, 0): def __init__(..., short_empty_elements): ... else: def __init__(...): ....

I thought there must be some standard for moving things from ``2and3`` when it founds they are actually not the same anymore.

**Comment by JelleZijlstra** [comment]
Let's keep it in 2and3. It's much better for maintenance to keep only a single copy of the code as long as the differences are minor.

**Comment by mcepl** [comment]
OK, what about this one?

File: stdlib/2and3/xml/sax/saxutils.pyi [hunk 1]
```diff
@@ -12,7 +14,11 @@ def unescape(data: str, entities: Mapping[str, str] = ...) -> str: ...
 def quoteattr(data: str, entities: Mapping[str, str] = ...) -> str: ...
 
 class XMLGenerator(handler.ContentHandler):
-    def __init__(self, out=..., encoding=...) -> None: ...
+    if sys.version_info > (3, 0):
```
Comment (JelleZijlstra) [inline stdlib/2and3/xml/sax/saxutils.pyi]: You should use `>=` here.

File: stdlib/2and3/xml/sax/saxutils.pyi [hunk 2]
```diff
@@ -12,7 +14,11 @@ def unescape(data: str, entities: Mapping[str, str] = ...) -> str: ...
...
+        def __init__(self, out=..., encoding=...,
+                     short_empty_elements: bool = False) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/xml/sax/saxutils.pyi]: The default should be `...`. Typeshed doesn't care what the value is.

**Comment by JelleZijlstra** [comment]
Thanks! I just fixed the merge conflict and I'll merge this in once Travis passes.

**Comment by mcepl** [comment]
so?

**Comment by JelleZijlstra** [comment]
Sorry, I got distracted before Travis was ready. Merging now.

**Comment by mcepl** [comment]
Thanks a lot.

**Pull request merged**